### DISCUSSION TITLE
x-for draggable issues

### DISCUSSION THREAD

**Discussion by liam-edwards** [alpinejs/alpine#1377] [7. Bugs (archived)]
Not sure how best to describe this issue, or even if it's an issue at all. Could very well be an issue with my code.

I have a list of items that gets rendered to the DOM within an `x-for` loop. I also have a way to drag each item in order to reorder them. All of this works some of the time however there are issues and I believe Alpine's `x-for` to be responsible for some of them, but please do correct me if I'm wrong.

Pen to reproduce:
https://codepen.io/liamedwards-dev/pen/eYvrrXO

In the pen, if you drag the fourth item to the second spot, depending on the speed of which you drag, one of three things will happen:
1. The fourth item won't move but the second and third will swap.
2. The fourth will move to the second spot, but the new third and fourth items will have swapped
3. Everything works correctly. This happens least of all for me.

**Comment by KevinBatdorf** [comment]
For me it seems to always work correctly. I tried dropping at different speeds, and tried various combinations but couldn't get it to error. Is it browser specific for you?
